enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,572 @@
+metadata:
+  name: mysnap
+spec:
+  volumeSnapshotClassName: csi-hostpath-snap [1] 
+  source:
+    persistentVolumeClaimName: myclaim [2]
+```
+
+1. The request for a particular class by the volume snapshot. If the **volumeSnapshotClassName** setting is absent 
+and there is a default volume snapshot class, a snapshot is created with the default volume snapshot class name. 
+But if the field is absent and no default volume snapshot class exists, then no snapshot is created.
+2. The name of the **PersistentVolumeClaim** object bound to a persistent volume. This defines what you want to 
+create a snapshot of. Required for dynamically provisioning a snapshot.
+
+#### VolumeSnapshotContent
+
+A snapshot taken of a volume in the cluster that has been provisioned by a cluster administrator.
+
+Similar to the PersistentVolume object, the VolumeSnapshotContent CRD is a cluster resource that
+points to a real snapshot in the storage back end.
+
+For manually pre-provisioned snapshots, a cluster administrator creates a number of
+VolumeSnapshotContent CRDs. These carry the details of the real volume snapshot in the storage system.
+
+The VolumeSnapshotContent CRD is not namespaced and is for use by a cluster administrator.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotContent
+metadata:
+  name: new-snapshot-content-test
+  annotations:
+    - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
+spec:
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
+  source:
+    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  sourceVolumeMode: Filesystem
+  volumeSnapshotRef:
+    name: new-snapshot-test
+    namespace: default
+```
+
+### Risks and Mitigations
+
+- Increased overhead.  The CSI snapshot controller and sidecar increase the on-disk footprint by a
+total of 360MB.  At idle, the components use a negligible amount of CPU (>1% CPU on a 2 core machine)
+and roughly 18Mb.
+
+### Drawbacks
+
+## Design Details
+
+### CSI Snapshotter Components
+
+The total additional cluster components are:
+
+1. VolumeSnapshot CRD
+2. VolumeSnapshotContent CRD
+3. VolumeSnapshotClass CRD
+
+### CSI-Snapshot Controller
+
+The CSI Snapshot Controller is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift.  The controller is a standalone component responsible for watching 
+`volumeSnapshot`, `volumeSnapshotContent`, and `volumeSnapshotClass` objects.  When a `volumeSnapshot` is created, 
+the controller will trigger a snapshot operation by creating a `volumeSnapshotConent` object, which will be 
+detected by the _CSI external-snapshotter sidecar_.
+
+MicroShift will embed the controller's manifests and deploy them during startup.  They are: 
+
+1. CSI VolumeSnapshot Controller Deployment
+2. CSI VolumeSnapshot Controller Service Account
+3. CSI VolumeSnapshot Controller ClusterRole and ClusterRoleBinding
+4. CSI VolumeSnapshot Controller Role and RoleBinding
```

Comment:
this should deploy also snapshot CRDs, https://github.com/openshift/csi-external-snapshotter/tree/master/client/config/crd

Reply:
I can see you mention the CRDs in MicroShift Assets below and basically repeat the same as here, only with file names. Is it useful?

Reply:
Thanks, good point. will fix.